data/schemas/keymap.jsonschema

```diff
@@ -53,5 +53,11 @@
             "type": "string",
             "description": "asdf"
         }
-    }
+    },
+    "required": [
+        "version",
```

Comment:
```
cat keyboards/clueboard/california/keymaps/default/keymap.json | qmk flash -
☒ Invalid JSON keymap: <_io.TextIOWrapper name='<stdin>' mode='r' encoding='utf-8'> : 'version' is a required property
```

Reply:
Also had to add defaulting to get stdin working on that keymap.json,
```diff
+++ b/lib/python/qmk/commands.py
@@ -185,6 +185,8 @@ def compile_configurator_json(user_keymap, bootloader=None, parallel=1, **env_va
 
         A command to run to compile and flash the C file.
     """
+    user_keymap["keymap"] = user_keymap.get("keymap", "default_json")
+
     # Write the keymap.c file
     keyboard_filesafe = user_keymap['keyboard'].replace('/', '_')
     target = f'{keyboard_filesafe}_{user_keymap["keymap"]}'
```

But that is maybe a different issue.

Reply:
Passing keymap.jsons directly from a keyboard dir to the CLI was not handled properly so far. These keymaps.jsons do not have to have the "keymap" argument as it can be inferred either from the arguments or the name of the keymap directory.
Added your fix to the PR.

Reply:
Regarding `version`, I was not entirely sure if we want it to be a hard requirement, but I think it'd make sense.
I can either remove the requirement or update the current `keymap.json` files, depending on what's preferred by others.

Reply:
As this is targeting master, and this field currently doesnt do much, my vote would be looser validation in the short term.

Reply:
Good point. Thinking about it this way, I have to agree.